legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java

```diff
@@ -461,23 +461,27 @@ public static Object evaluate(ExecutionSupport es, org.finos.legend.pure.m3.core
         }
         if (func instanceof ConcreteFunctionDefinition)
         {
-            JavaMethodWithParamsSharedPureFunction<?> p = (JavaMethodWithParamsSharedPureFunction<?>) getSharedPureFunction(func, bridge, es);
-            Class<?>[] paramClasses = p.getParametersTypes();
-            int l = paramClasses.length;
-            MutableList<Object> paramInstances = FastList.newList();
-            for (int i = 0; i < (l - 1); i++)
-            {
-                if (instances[i] instanceof RichIterable && paramClasses[i] != RichIterable.class)
+            SharedPureFunction<?> pureFunc = getSharedPureFunction(func, bridge, es);
+            if (pureFunc instanceof JavaMethodWithParamsSharedPureFunction) {
+                JavaMethodWithParamsSharedPureFunction<?> p = (JavaMethodWithParamsSharedPureFunction<?>) pureFunc;
+                Class<?>[] paramClasses = p.getParametersTypes();
+                int l = paramClasses.length;
+                MutableList<Object> paramInstances = FastList.newList();
+                for (int i = 0; i < (l - 1); i++)
                 {
-                    paramInstances.add(CompiledSupport.toOne(((RichIterable<?>) instances[i]), null));
-                }
-                else
-                {
-                    paramInstances.add(instances[i]);
+                    if (instances[i] instanceof RichIterable && paramClasses[i] != RichIterable.class)
+                    {
+                        paramInstances.add(CompiledSupport.toOne(((RichIterable<?>) instances[i]), null));
+                    }
+                    else
+                    {
+                        paramInstances.add(instances[i]);
+                    }
                 }
+                return p.execute(paramInstances, es);
+            } else {
+                pureFunc.execute(Lists.fixedSize.with(instances), es);
```

Comment:
Don't you still need the same type of parameter checking as above (i.e., lines 472-479)?

Reply:
I think this conversion happens within the SharedPureFunction implementation, but will double check

Reply:
The parameter adapting seems specific to JavaMethodWithParamsSharedPureFunction (I assume because it needs to be fully type correct for invoking the method where others can handle item or [item]).  Perhaps in a future change it should be moved in to the JavaMethodWithParamsSharedPureFunction execute implementation.